### PR TITLE
remove redundant sensor fetch and bump 0.2.10 -> 0.2.11

### PR DESCRIPTION
**Pull request by devProdigy**
## Description

Time execution optimization by removing redundant sensor data call.

[Azure DevOps Link](https://dev.azure.com/EquinorASA/OMNIA%20Prevent/_workitems/edit/17069)

## Type of change

Mark the relevant options (`[x]` will mark checkbox as checked).
- [ ] Bugfix
- [X] Feature
- [ ] Code style update (formatting, local variables)
- [X] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [X] Tests
- [ ] Other

# Checklist:

- [ ] I have made corresponding changes to the documentation / README (if needed)
- [X] I have added tests that prove my fix is effective or that my feature works (if needed)

### CODE REVIEW

File: tests/unit/test_executor.py [hunk 1]
```diff
@@ -1,12 +1,24 @@
...
+@patch("latigo.executor.PredictionExecutor._perform_auth_checks", new=MagicMock())
+def test__execute_prediction_success(config):
+    executor = PredictionExecutor(config=config)
+    task = TaskFactory()
+    revision = "000"
 
-def test_fetch_task(config):
-    pe = PredictionExecutor(config=config)
+    client_mock = MagicMock(name="client_mock")
+    with patch.object(executor.prediction_executor_provider.gordo_pool, "allocate_instance", return_value=client_mock):
```
Comment (o-fedorov) [inline tests/unit/test_executor.py:19]:
```suggestion
    with patch.object(executor.prediction_executor_provider.gordo_pool, "allocate_instance") as mock_pool:
        client_mock = mock_pool.return_value
```

File: tests/unit/test_executor.py [hunk 2]
```diff
@@ -1,12 +1,24 @@
...
+        executor._execute_prediction(task=task, revision=revision)
```
Comment (o-fedorov) [inline tests/unit/test_executor.py:20]: Tasting private methods looks wrong. It seems that `GordoPredictionExecutionProvider.execute_prediction` should be tested instead.

File: tests/unit/test_executor.py [hunk 3]
```diff
@@ -1,12 +1,24 @@
...
+from latigo.gordo import GordoModelInfoProvider
+from tests.factories.task import TaskFactory
 
-logger = logging.getLogger(__name__)
 
+@patch("latigo.gordo.prediction_execution_provider.GordoClientPool", new=MagicMock())
+@patch(
+    "latigo.executor.model_info_provider_factory", new=MagicMock(side_effect=MagicMock(spec_set=GordoModelInfoProvider))
+)
+@patch("latigo.executor.PredictionExecutor._perform_auth_checks", new=MagicMock())
```
Comment (o-fedorov) [inline tests/unit/test_executor.py:12]:
```suggestion
@patch("latigo.gordo.prediction_execution_provider.GordoClientPool")
@patch(
    "latigo.executor.model_info_provider_factory", side_effect=MagicMock(spec_set=GordoModelInfoProvider))
)
@patch("latigo.executor.PredictionExecutor._perform_auth_checks")
```
Default `new` argument is `MagicMock`. Also not sure what is the purpose of additional arguments to the patch of `"latigo.executor.model_info_provider_factory"`. Perhaps plain `@patch("latigo.executor.model_info_provider_factory", spec=GordoModelInfoProvider)` is enough.

Comment (o-fedorov) [reply tests/unit/test_executor.py:12]: Also patching a private method looks wrong

File: tests/unit/test_executor.py (hunk 1 above)
Comment (devProdigy) [reply tests/unit/test_executor.py:19]: Thanks, but this won't work cause new object will be created later in the code.

File: tests/unit/test_executor.py (hunk 2 above)
Comment (devProdigy) [reply tests/unit/test_executor.py:20]: it is tested in scope of this test but that "private" method should be tested as well. Maybe it should not be private at all

File: tests/unit/test_executor.py (hunk 3 above)
Comment (devProdigy) [reply tests/unit/test_executor.py:12]: if you won't specify `new` then you should pass it as the argument and we don't need it in the function.
Plain won't work cause we're mocking a factory here.

Can't see nothing wrong in the private method mocking, sometimes like here you have to disable some inner logic and this is the way to do it. Why is it wrong from your perspective?

**Pull request merged**